### ISSUE TITLE
Add a second Game Path option in the Config to allow us to install mods in the My Docs Data folder instead of Starfield's root Data folder as needed.

### ISSUE THREAD

**Issue by MrDenimLP** [ManApart/starfield-mod-manager#1]
For reference, I installed the Galactic Civil War Conversion and it's retexure through the tool. The file structure was correct (had to change that manually), however it installed the mod in Starfield's Root Data folder (SteamLibrary/steamapps/common/Starfield/Data/), not in the My Documents/My Games/Starfield/Data as it's manual install requires.

I think it would be a good idea to use a second game-path in the Config - or let us use the ini-path as that's the same location - to give us the option where we want to install a particular mod, as some mods still require their files to be installed in the Data folder in the My Docs, not the Steam Library Root.

EDIT: I don't know how to add labels to the title, sorry about the triple update.

**Comment by ManApart** [comment]
Hey MrDenimLP, thanks for the issue report.

Could you try manually placing the files in my documents and see if it works?

If I remember correctly, I had issues with wine + using the my documents approach.

In my manual testing before building this tool, the best/only way I got mods to work on linux was with SFSE + [disable my docs plugin](https://www.nexusmods.com/starfield/mods/1599). This has the added benefit that you can continue to use custom loading screens from your in game screenshots.

**Comment by MrDenimLP** [comment]
1. That's what I wound up doing and it worked fine.
2. Does symlinking mods files to the My Docs in the comptdata not work properly like it does in the Starfield root folder?
3. I haven't gotten around to trying SFSE and such dependent mods yet. I'll let you know if I have any issues.

I know this is still in heavy development, but I was planning on typing up a quick how-to and pushing this tool on Reddit. I already posted about it on r/linux_gaming. I'll update that post and make one on r/Starfield as well. I really want this tool to succeed because it's the only one that's been able to successfully help me mod Starfield. The various other methods like Steam Tinker Launch's/Lutris' deployment of Vortex and MO2 and manually installing those tools into Starfield's comptdata folder never worked for me.

**Comment by ManApart** [comment]
1) Ok, I'll see what I can do  to support linking to the documents in mycompdata. It's _possible_ it would just work if you point the game data to that dir, but then would fail for things that _need_ the game directory, like the sfse install. I'll have to look at it, but can maybe just add a second link for docs, and if you don't give one it just uses the game dir. Would you be willing to  test and let me know if it works, once I have something pushed up?

2) In my testing I didn't think it worked, but I also prefer the game data path so I probably wasn't very thorough. If it works for you I can add it as an option for non SFSE users.

3) Sounds good

4) Regarding reddit, my plan was to post there once I was ready to release, but I don't have an account, so if you're willing to post it to different subs, that's great. That said, I'd like to do that once I'm at 1.0 / published on the nexus. I'm actually in the specific niche of Linux+Nvidia that can't play the game yet. I'm hoping to wait to release this until the drivers are updated and I can actually play on linux so I can better battle test it. So it may be a little bit yet, depending on drivers.

Would you be willing to send me the reddit links so I can see peoples reactions?

Ps. I appreciate the encouragement. I  updated the readme with all the linux starfield modding tools I had found before starting on this app. The lutris/ wine stuff people were doing was buggy for me and the existing managers didn't do what I was personally hoping for. I think a CLI tool won't be for everyone, so it's nice to hear at least one person is excited about it. Thank you!

**Comment by ManApart** [comment]
Regarding fomods:

Fomods are currently "compatible" but not "supported".

When you install a fomod, it will stage it and then warn you (validation gives the same warning about all installed mods, if they have issues or are fomods etc).

Parsing the fomod would be a decent lift and I don't think provides a ton of value, as the user experience of installing a fomod through cli is probably not any better than just opening the local folder and choosing options in the file explorer (which is what the warning instructs you to do).

So basically you just say `local 5`  to open the fomod staging folder, and then delete the options you don't want etc, and then what's left will be deployed. It's not ideal, but its serviceable for now. I'm open to better options but it'd be worth its own issue.

**Comment by MrDenimLP** [comment]
I'm also an Nvidia user on the 545 drivers from nvidia-all and I don't know if I'd call it "Ghosting" or what but character models and buildings... blur or shake is how I can best describe it. I can barely manage 30fps on my laptop's 2070 Super and the "Ghosting" gives me a headache. If I had the money I'd splurge on a high-end AMD laptop in a heartbeat.

**Comment by MrDenimLP** [comment]
Here are the reddit links:
[r/linux_gaming](https://www.reddit.com/r/linux_gaming/comments/17lmq61/i_found_a_cli_mod_manager_for_starfield_and_wrote/)
[r/Starfield](https://www.reddit.com/r/Starfield/comments/17lmom6/i_found_a_cli_mod_manager_for_us_linux_users_and/)
[r/LowSodiumStarfield](https://www.reddit.com/r/NoSodiumStarfield/comments/17lmpaf/i_found_a_cli_mod_manager_for_us_linux_users_and/)

**Comment by ManApart** [comment]
Thank you!

Per your comment, last night I tried the 545 drivers and Starfield is now playable on my linux machine! Once I have time I'll do more testing on the mod manager now that I can actually get past the main menu.

That reddit post is great! Would you be willing to submit that as a PR to something like `setup-guide.md` so we can include it in this repo?

I've pushed to master a "useMyDocs" option. Would you be willing to pull it down, build, and see if it works for you? It re-uses the ini-path and testing with some test folders it looks good to me.

**Comment by MrDenimLP** [comment]
I'll take a look after class and work.

**Comment by MrDenimLP** [comment]
How do I go about building the new master? My CLI knowledge for Git is limited to cloning a repo and `makepkg -si`. That doesn't work with this repo because there is no PKGBUILD.

**Comment by ManApart** [comment]
If you're in the repo, run `./gradlew build jar`.

The jar should be placed in `build/libs`.

If that doesn't work, let me know and I'll push a new tagged version / release.

**Comment by MrDenimLP** [comment]
So! Initial testing proved successful. I purged my mod list with the original smm.desktop and removed the launch option from Steam for sfse_loader.exe. I installed the Galactic Civil War Conversion with the new setup and it did install to the My Docs folder as I checked the flag. It also did not fully purge when I used the command. It deleted the individual files but the folder structure was still there. The Kill My Games Folder mod doesn't want a Data folder there at all so if I did not check that before resetting back to the original setup, I would have had to go all the way into the My Doc folder to manually delete the Data folder that was in there.

The _only_ concern with this that I have is that there was no selection system in place. You know, something that could give me the option of where I want to install the mod. Something like this, for example:

`Where do you want to install <mod_name>?`
`[1] Starfield Root directory   [2] Starfield My Documents folder`
`[user@distro ~]$ 2`
`Are you sure? [y/n]`
`[user@distro ~]$ Y`

This way you could do away with the manual flagging with `config use-my-docs true/false` and just let the folks that haven't yet - or don't want to - install the Kill My Games Folder mod get right to work with less configuration on the front-end.

**Comment by ManApart** [comment]
I think the issues with per mod deploys are:
1) You have to make the decision every time instead of just once. (You can toggle the my-docs config at any time)
2) I don't believe the two deployments are compatible. If you turn on the my docs way, it won't recognize anything in the game folder. So I don't think it _can_ be per mod.

I'll look into purging the empty folders, though that may be tricky to do safely.

**Comment by MrDenimLP** [comment]
What if they did partial deployments? Like, enable and deploy the root folder mods first, then turn on the UseMyDocs switch then enable and deploy the My Docs mods.

It's getting kind of late here so I'll give that a shot tomorrow.

**Comment by MrDenimLP** [comment]
Posted the guide on [r/SteamDeck](https://www.reddit.com/r/SteamDeck/comments/17mnmjj/i_found_a_cli_mod_manager_for_starfield_and_wrote/) as well.

**Comment by ManApart** [comment]
That could possibly work, though personally I don't think that use case is worth the complexity to support. I also don't want benefit the 1% if it adds burden (picking target for every deploy) to the majority of users.
If that's really important to someone they could keep two copies of the jar and use one to manage root mods and the other to do docs.

I think in the great majority of cases people will want to use SFSE and/or the game path deployment, as that's how you deploy to all the other BGS games and many mods use SFSE.

So I think the main target should be the game path, and having an option to deploy everything to  the ini path is a good compromise for people not yet ready (or unwilling) to move to the main deploy path.

**Comment by ManApart** [comment]
Purging empty folders has been added to master. Could you please pull latest and let me know if that is working and the my docs is working? If so I'd like to close this issue.

Also, would you be willing to submit that as a PR to something like `setup-guide.md` so we can include it in this repo?, (or do you mind if I add it)? That way a how to guide is part of the repo.

**Comment by MrDenimLP** [comment]
Go ahead and add it yourself. I tried to do it when you first mentioned it but I honestly couldn't figure out how.

**Comment by ManApart** [comment]
Ok sounds good. :)
